src/CoreFoundation/DispatchData.cs

```diff
@@ -88,6 +88,18 @@ public static DispatchData FromBuffer (IntPtr buffer, nuint size)
 			return new DispatchData (dd, owns: true);
 		}
 
+		// create a dd using the data of the span 
+		public static DispatchData FromReadOnlySpan (ReadOnlySpan<byte> content)
+		{
+			unsafe {
+				fixed (byte* ptr = content) {
+					// set the block that will be used to clean the data once we are done with it
+					var dd = dispatch_data_create ((IntPtr) ptr, (nuint) content.Length, IntPtr.Zero, destructor: IntPtr.Zero);
+					return new DispatchData (dd, owns: true);
```

Comment:
I think this leaks memory, because nobody will release the pointer passed to dispatch_data_create.

Reply:
Sorry, I phrased that wrong, what I meant is that the pointer is a pointer to stack memory, and as such it will be invalid as soon as this function returns, but the DispatchData still contains it and may use it later.

However, after re-reading the documentation for `dispatch_data_create`, it seems the data the pointer points to is copied internally when specifying DISPATCH_DATA_DESTRUCTOR_DEFAULT as the destructor, and DISPATCH_DATA_DESTRUCTOR_DEFAULT=NULL, which is what you're passing, which means the code is OK. A comment might be nice though for the next people wondering if `dispatch_data_create` keeps the pointer around or not.